Add information about reviewing Dependabot PRs (#13)
alphagov/design-system-team-docs: development/code-reviews.md

Update the relevant Trello ticket for the PM to review the change (or should
this happen before the merge?)

## When you review a Dependabot PR

> You should check that the pull request:

Reviewer: ```suggestion
You should make sure the pull request:
```

 - works correctly

>  - does not introduce unexpected or undesired functionality

Reviewer: ```suggestion
 - does not introduce unexpected or unwanted functionality
```

> After you have approved the pull request, ask another developer from the team to review it. After the other developer has approved the pull request, they merge it.

Reviewer: ```suggestion
To make sure the pull request works correctly:

1. Look at where the package is used with `npm ls`.
2. Test that the features using the package work correctly.

After you have approve the pull request, ask another developer from the team to review it. After the other developer approves the pull request, they can merge it.
```

Update the relevant Trello ticket for the PM to review the change (or should
this happen before the merge?)

## When you review a Dependabot PR

To make sure the pull request works correctly:

- Read the release notes for the new version and look for changes that would impact our codebase. This is especially important with major releases.
- Look at where in our codebase the package is used with `npm ls` - [see this example, where tests, Rollup and linting used the updated package](https://github.com/alphagov/govuk-design-system/pull/1813#issuecomment-896654240). Test that the parts of the codebase using the package work correctly.
- If the new version is a major release, consider waiting for a few days before updating to see if anyone reports issues that could impact us.

> After you approve the pull request, ask another developer from the team to review it. After the other developer approves the pull request, they can merge it.

Reviewer 2: Is it worth making this more generic - i.e: the person reading this documentation may already be the second developer reviewing the dependabot PR, so they wouldn't go and ask someone else. Maybe it's enough to say that dependabot PRs should be treated like external contributions and require 2 approving reviews before merging?

Author: Updated, thanks to @EoinShaughnessy for word wrangling help.
